lib/facilities/bulk_client.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+require 'facilities/gis_middleware'
+
+# Client class for doing periodic bulk refreshes of GIS data.
+# Allows inspection of last modified date to update only when needed.
+module Facilities
+  class BulkClient
+    BATCH_SIZE = 1000.0
+    OPEN_TIMEOUT = 2
+    REQUEST_TIMEOUT = 6
+
+    COUNT_PARAMS = {
+      where: '1=1',
+      returnCountOnly: true,
+      f: 'json'
+    }.freeze
+
+    ALL_PARAMS = {
+      where: '1=1',
+      inSR: 4326,
+      outSR: 4326,
+      returnGeometry: true,
+      outFields: '*',
+      f: 'json'
+    }.freeze
+
+    def initialize(url)
+      @url = url
+      @conn = Faraday.new(url: @url) do |conn|
+        conn.options.open_timeout = OPEN_TIMEOUT
+        conn.options.timeout = REQUEST_TIMEOUT
+        conn.response :gis_json
+        # conn.response :logger
+        # TODO conn.use :breaker
+        conn.adapter Faraday.default_adapter
+      end
+    end
+
+    def last_edit_date
+      response = @conn.get('', f: 'json')
+      response.body.dig('editingInfo', 'lastEditDate')
+    rescue Faraday::ConnectionFailed => e
+      Rails.logger.warn "Facility lastEditDate check connection failed: #{e.message}"
+      return nil
+    rescue Faraday::TimeoutError
+      Rails.logger.warn 'Facility lastEditDate check timeout'
+      return nil
+    rescue Facilities::Errors::ServiceError => e
+      Rails.logger.warn "Facility lastEditDate check request failed: #{e.message}"
+      return nil
+    end
```

Comment:
Do you really need all these Rails.logger calls? If a Faraday exception is raised isn't whatever is provided in Sentry sufficient to debug the issue?